### PR TITLE
Dialog causes crash when put in background (EXPOSUREAPP-14569)

### PR DESCRIPTION
**Pull request by SamuraiKek**
[Ticket.](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14569)

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/repository/SrsSubmissionRepository.kt [hunk 1]
```diff
@@ -164,8 +164,7 @@ class SrsSubmissionRepository @Inject constructor(
                 val errorCode = e.type.toSrsErrorType()
                 SrsSubmissionException(
                     errorCode = errorCode,
-                    errorArgs = errorCode.errorArgs(appConfig.selfReportSubmission),
-                    cause = e
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/repository/SrsSubmissionRepository.kt:168]: Passing the cause will lead to `SrsSubmissionException` parcelization errors. We aren't using the cause to display error dialogs anyway.

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/repository/SrsSubmissionRepository.kt:168]: would `@IgnoredOnParcel` prevent the crash?

Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/repository/SrsSubmissionRepository.kt:168]: I'll give it a try

Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/repository/SrsSubmissionRepository.kt:168]: Update: nope, still crashing with `@IgnoredOnParcel`

**Review by mtwalli — APPROVED** [review]

**Pull request merged**